test/unit/uk/gov/hmrc/exports/services/mapping/goodsshipment/PreviousDocumentsBuilderSpec.scala

```diff
@@ -48,6 +48,69 @@ class PreviousDocumentsBuilderSpec extends UnitSpec with ExportsDeclarationBuild
 
       }
 
+      "a MUCR has been specified but no other PreviousDocument elements have" in {
+        val builder = new PreviousDocumentsBuilder
+        val mucr = MUCR(VALID_MUCR)
+        val goodsShipment = new GoodsShipment
+        builder.buildThenAdd(mucr, goodsShipment)
+
+        val previousDocs = goodsShipment.getPreviousDocument
+        previousDocs.size must be(1)
+        previousDocs.get(0).getID.getValue must be(VALID_MUCR)
+        previousDocs.get(0).getCategoryCode.getValue must be("Z")
+        previousDocs.get(0).getTypeCode.getValue must be("MCR")
+        previousDocs.get(0).getLineNumeric must be(BigDecimal(1).bigDecimal)
+      }
+
+      "a MUCR has been specified along with other PreviousDocument elements" when {
```

Comment:
Is it worth adding one more test case when there is MUCR and DUCR?

Reply:
Well my thinking is that a DUCR previous document element is no different to any other type of previous document element. There is no special status to a DUCR previous document element that if combined with a MUCR previous document element would cause the code any particular issue, so I'm not convinced of what value the test would add.

Reply:
It is a bit different, as the "entry point" is different - DUCR is taken from `ConsignmentReferences`.
The value of this test case would be to test one of the overloaded methods. 

I'll leave it to your opinion, but I'd add this test case to also cover all potential scenarios.